### PR TITLE
#330 - Update MS Update to 2021.01.xx

### LINKED ISSUE #330: MS Update to 2021.01.xx
### Description
We have to consider (check/test) that the Thematic tool must continue to work as it is along with recent updates we provided for the visual style editor.

Involved MS developments for this project update are mainly related to the following issues, requested by c040:

- [x] https://github.com/geosolutions-it/MapStore2/issues/5930
- [x] https://github.com/geosolutions-it/MapStore2/issues/6101
- [x] https://github.com/geosolutions-it/MapStore2/issues/5883: the Query panel should reports the same measure of the measurement tool. To check the geodetic property setting, now it is set to false: both should be the same to true
- [x] https://github.com/geosolutions-it/MapStore2/issues/6102
- [x] https://github.com/geosolutions-it/MapStore2/issues/2691
- [x] https://github.com/geosolutions-it/MapStore2/issues/4150
- [x] https://github.com/geosolutions-it/MapStore2/issues/6103
- [x] https://github.com/geosolutions-it/MapStore2/issues/6338
- [x] https://github.com/geosolutions-it/MapStore2/issues/6358
- [x] https://github.com/geosolutions-it/MapStore2/issues/6316

Also some other dependencies versions must be updated according to the new MS version used: see as a reference https://github.com/geosolutions-it/MapStore2/issues/6316

### Other useful information (optional):
To evaluate the update using the stable branch 2021.01.xx of MapStore.

### PR DESCRIPTION
**Pull request by MV88**
## Description
Update MS Update to 2021.01.xx

## Issues
 - Fix #330

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [x] Other... Please describe: ms2 update

**What is the current behavior?** (You can also link to an open issue here)
ms2 outdated

**What is the new behavior?**
ms2 updated

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes
 - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: prod-webpack.config.js [hunk 1]
```diff
@@ -12,18 +12,18 @@ const paths = {
 
 const config = require('./MapStore2/build/buildConfig')(
```
Comment (offtherailz) [inline prod-webpack.config.js:13]: I see the module federation plugin is missing here. I think it is needed to support the extensions, as far as in
https://mapstore.readthedocs.io/en/latest/developer-guide/mapstore-migration-guide/

(we have the UI for installing extensions, isn't it?)

File: webpack.config.js [hunk 2]
```diff
@@ -3,10 +3,10 @@ const extractThemesPlugin = require('./MapStore2/build/themes.js').extractThemes
...
 const config = require('./MapStore2/build/buildConfig')(
     {
-        'MapStore2-C040': path.join(__dirname, "js", "app"),
-        "embedded": path.join(__dirname, "js", "embedded"),
-        "ms2-api": path.join(__dirname, "js", "api"),
-        "llpp": path.join(__dirname, "js", "llpp")
+        'MapStore2-C040': path.join(__dirname, "js", "apps", "mapstore"),
+        "embedded": path.join(__dirname, "js", "apps", "embedded"),
+        "ms2-api": path.join(__dirname, "js", "apps", "api"),
+        "llpp": path.join(__dirname, "js", "apps", "llpp")
```
Comment (offtherailz) [inline webpack.config.js:9]: I see the module federation plugin is missing here. I think it is needed to support the extensions, as far as in
https://mapstore.readthedocs.io/en/latest/developer-guide/mapstore-migration-guide/

(we have the UI for installing extensions, isn't it?)

File: js/app.jsx [hunk 3]
```diff
@@ -6,7 +6,7 @@
  * LICENSE file in the root directory of this source tree.
 */
 
-require('../MapStore2/web/client/product/main')(
+require('../MapStore2/web/client/product/main').default(
```
Comment (offtherailz) [inline js/app.jsx:9]: I can suggest to migrate also this file to imports, because all the other that include it are migrated.

**Review by offtherailz — CHANGES_REQUESTED** [review]
Only a couple of thing from migration guide that are missing. They are not strictly mandatory, but part of the procedure, so I think they should be aligned (to avoid future issues due to procedure not correctly applied).

**Comment by MV88** [comment]
pr updated, the extension UI is present

image

**Review by offtherailz — APPROVED** [review]
Let's go to test in test env :+1:

**Pull request merged**